### DISCUSSION TITLE
How to create a Bot that has Tabs in it using teams-toolkit version 5?

### DISCUSSION THREAD

**Discussion by IAM5K** [microsoft/hack-together-teams#38] [Q&A]
We were developing an app that had a command bot and tabs to show the dashboard and settings.
Something similar to this:
![image](https://github.com/microsoft/hack-together-teams/assets/43265495/c4e02745-b89c-4e9a-93fb-ebc9c2562733)

My question is,
Earlier in Teams toolkit version 4 we had the option to create a bot and add feature to it. Where we can add tabs to it.
Is it still possible to add tabs to a bot ? And if no then are you planing to remove this from app feature?
If it is still possible, please share how can we achieve that as we do not have a add feature option there in teams toolkit version 5.

**Comment by aycabas** [comment]
Hi @IAM5K, in Teams Toolkit for VS Code V5, we encourage developers to learn the processes behind the scenes. So Add feature capability is replaced with "View how to guides". If you select the "configure Tab app capability", you'll be directed to the following page and you can follow the steps to add tab capability in your bot: https://github.com/OfficeDev/TeamsFx/wiki/How-to-configure-Tab-capability-within-your-Teams-app

![image](https://github.com/microsoft/hack-together-teams/assets/36196437/7266c419-351a-4d5a-88b2-187d91c1814f)

Let me know if you have any other questions along the way.

**Comment by IAM5K** [comment]
![image](https://github.com/microsoft/hack-together-teams/assets/43265495/55c8b1ba-0785-470b-93e0-7df07c38e54c)

Document is unclear about generating the tab folder, if we generate tab in the folder then there is project inside bot, which doesn't matches with the sample mentioned above.

My Directory structure, 
![image](https://github.com/microsoft/hack-together-teams/assets/43265495/c1643bac-27ee-482f-9ce3-8d1aa4bec14b)

Directory structure of example given
![image](https://github.com/microsoft/hack-together-teams/assets/43265495/60e57c26-2109-41df-9a59-314922b2b1f2)

**Reply by aycabas** [reply]
Hi @IAM5K, as it shows in the example structure, you do not need to have all the resources under your tab folder such as `teamsapp.local.yml`, `teams.app.yml` etc. All the resources will be under the main root and used by both Bot and Tab. The tab folder will only have `public`, `src` folders, `package.json` and `tsconfig.json` files.

As you can see in your project structure, you have dublicate of teamsapp.local.yml, .localConfigs etc. That makes your tab project separate than your bot project. You'll have to use the same configuration files. 

Your project structure is supposed to look as the following:

env/
appPackage/
infra/
**tab/ (Your tab code)**
--public/
-- src/
-- tsconfig.json
-- package.json
**src/ (your existing bot source code, you can rename this folder as *bot*)**
-- adaptivecards/
-- index.ts
-- tsconfig.json
-- package.json
-- ... (other bot files)
teamsapp.yml
teamsapp.local.yml
README.md
.gitignore
.
.
.

I hope this helps. Let me know if it works.

**Reply by dhaneshb** [reply]
How did you mange to create this structure aycabas? when i create a bot app in the tab project, it created as a seperate project and has a lot of duplicates. Did you remove them and organize them manually?

**Reply by IAM5K** [reply]
@dhaneshb 
![image](https://github.com/user-attachments/assets/842b667b-4d28-4f9f-9546-b6ba8d58a370)
there has to be lot of manual work to rename and configure all paths.
